doc.go

```diff
@@ -18,5 +18,92 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 // THE SOFTWARE.
 
-// Package zap provides fast, structured, leveled logging in Go.
+// Package zap provides fast, structured, leveled logging.
```

Comment:
do we have a good link that can explain leveled logging?  Wouldn't be surprised if new engineers are not as familiar with the nuances.